sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -794,6 +794,13 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val FAIL_AMBIGUOUS_SELF_JOIN =
```

Comment:
Would `FAIL_AMBIGUOUS_DATASET_COLUMN_REF` be more accurate?
And in the doc below, we can describe when this usually happens (e.g., with self-join).

Reply:
A self-join that contains ambiguous column reference is an ambiguous self-join. I think ambiguous self-join is shorter and easier to understand, as a config name.

Reply:
BTW there are no other places that can make column reference ambiguous. Self-join is the only place.